psi4/driver/p4util/fcidump.py

```diff
@@ -0,0 +1,209 @@
+
+    """
+    # Get some options
+    reference = core.get_option('SCF', 'REFERENCE')
+    ints_tolerance = core.get_option('SCF', 'INTS_TOLERANCE')
+    # Some sanity checks
+    if reference not in ['RHF', 'UHF']:
+        raise ValidationError('FCIDUMP not implemented for {} references\n'.format(reference))
+    if oe_ints is None:
+        oe_ints = []
+
+    molecule = wfn.molecule()
+    docc = wfn.doccpi()
+    frzcpi = wfn.frzcpi()
+    frzvpi = wfn.frzvpi()
+    active_docc = docc - frzcpi
+    active_socc = wfn.soccpi()
+    active_mopi = wfn.nmopi() - frzcpi - frzvpi
+
+    nbf = active_mopi.sum() if wfn.same_a_b_orbs() else 2 * active_mopi.sum()
+    nirrep = wfn.nirrep()
+    nelectron = 2 * active_docc.sum() + active_socc.sum()
+
+    core.print_out('Writing integrals in FCIDUMP format to ' + fname + '\n')
+    # Generate FCIDUMP header
+    header = '&FCI\n'
+    header += 'NORB={:d},\n'.format(nbf)
+    header += 'NELEC={:d},\n'.format(nelectron)
+    header += 'MS2={:d},\n'.format(wfn.nalpha() - wfn.nbeta())
+    header += 'UHF=.{}.,\n'.format(not wfn.same_a_b_orbs()).upper()
+    orbsym = ''
+    for h in range(active_mopi.n()):
+        for n in range(frzcpi[h], frzcpi[h] + active_mopi[h]):
+            orbsym += '{:d},'.format(h + 1)
+            if not wfn.same_a_b_orbs():
+                orbsym += '{:d},'.format(h + 1)
+    header += 'ORBSYM={}\n'.format(orbsym)
+    header += '&END\n'
+    with open(fname, 'w') as intdump:
+        intdump.write(header)
+
+    # Get an IntegralTransform object
+    check_iwl_file_from_scf_type(core.get_option('SCF', 'SCF_TYPE'), wfn)
+    spaces = [core.MOSpace.all()]
+    transType = core.IntegralTransform.TransformationType.Restricted
+    if not wfn.same_a_b_orbs():
+        transType = core.IntegralTransform.TransformationType.Unrestricted
+    ints = core.IntegralTransform(wfn, spaces, transType)
+    ints.transform_tei(core.MOSpace.all(), core.MOSpace.all(), core.MOSpace.all(), core.MOSpace.all())
+    core.print_out('Integral transformation complete!\n')
+
+    DPD_info = {'instance_id': ints.get_dpd_id(), 'alpha_MO': ints.DPD_ID('[A>=A]+'), 'beta_MO': 0}
+    if not wfn.same_a_b_orbs():
+        DPD_info['beta_MO'] = ints.DPD_ID("[a>=a]+")
+    # Write TEI to fname in FCIDUMP format
+    core.fcidump_tei_helper(nirrep, wfn.same_a_b_orbs(), DPD_info, ints_tolerance, fname)
+
+    # Read-in OEI and write them to fname in FCIDUMP format
+    # Indexing functions to translate from zero-based (C and Python) to
+    # one-based (Fortran)
+    mo_idx = lambda x: x + 1
+    alpha_mo_idx = lambda x: 2 * x + 1
+    beta_mo_idx = lambda x: 2 * (x + 1)
+
+    with open(fname, 'a') as intdump:
+        core.print_out('Writing frozen core operator in FCIDUMP format to ' + fname + '\n')
+        if reference == 'RHF':
+            PSIF_MO_FZC = 'MO-basis Frozen-Core Operator'
+            moH = core.Matrix(PSIF_MO_FZC, wfn.nmopi(), wfn.nmopi())
```

Comment:
Hmm, does the Wavefunction `H` received the frozen core contributions downstream? I work a bit the our IWL computers do not take into account one-electron perturbations.

Reply:
Where/how can I check if one-electron perturbations are included?